### PR TITLE
feat: add custom context to subgraph checks

### PR DESCRIPTION
**Pull request by JivusAyrus**
## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: connect/src/wg/cosmo/platform/v1/platform_pb.ts [hunk 1]
```diff
@@ -774,6 +774,55 @@ export class GitInfo extends Message<GitInfo> {
   }
 }
 
+/**
+ * @generated from message wg.cosmo.platform.v1.CustomCheckContext
+ */
+export class CustomCheckContext extends Message<CustomCheckContext> {
```
Comment (StarpTech) [inline connect/src/wg/cosmo/platform/v1/platform_pb.ts:780]: I'd give it a concrete name. What about VcsContext?

File: cli/src/commands/subgraph/commands/check.ts [hunk 2]
```diff
@@ -39,6 +40,15 @@ export default (opts: BaseCommandOptions) => {
     }
 
     const { gitInfo, ignoreErrorsDueToGitHubIntegration } = await verifyGitHubIntegration(opts.client);
+    let customContext: CustomCheckContext | undefined;
+
+    if (config.checkAuthor || config.checkCommitSha || config.checkBranch) {
```
Comment (StarpTech) [inline cli/src/commands/subgraph/commands/check.ts:45]: Will we now use one mechanism to provide VCS integration or do we distinguish?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: connect/src/wg/cosmo/platform/v1/platform_pb.ts (hunk 1 above)
Comment (JivusAyrus) [reply connect/src/wg/cosmo/platform/v1/platform_pb.ts:780]: Done

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**